crates/common/checkpoint_sync/src/lib.rs

```diff
@@ -0,0 +1,147 @@
+pub mod checkpoint;
+
+use alloy_primitives::B256;
+use anyhow::{anyhow, ensure};
+use checkpoint::get_checkpoint_sync_sources;
+use ream_consensus::{
+    blob_sidecar::{BlobIdentifier, BlobSidecar},
+    constants::SECONDS_PER_SLOT,
+    electra::{beacon_block::SignedBeaconBlock, beacon_state::BeaconState},
+    execution_engine::rpc_types::get_blobs::BlobAndProofV1,
+};
+use ream_fork_choice::{handlers::on_tick, store::get_forkchoice_store};
+use ream_network_spec::networks::network_spec;
+use ream_storage::{db::ReamDB, tables::Table};
+use reqwest::{
+    Url,
+    header::{ACCEPT, HeaderValue},
+};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use ssz::Decode;
+use tracing::{info, warn};
+
+/// A OptionalBeaconVersionedResponse data struct that can be used to wrap data type
+/// used for json rpc responses
+///
+/// # Example
+/// {
+///  "data": json!({
+///     "version": Some("electra")
+///     "execution_optimistic" : Some("false"),
+///     "finalized" : None,
+///     "data" : T
+/// })
+/// }
+#[derive(Debug, Serialize, Deserialize)]
+pub struct OptionalBeaconVersionedResponse<T> {
+    pub version: Option<String>,
+    #[serde(default)]
+    pub execution_optimistic: Option<Value>,
+    #[serde(default)]
```

Comment:
```suggestion
    pub execution_optimistic: Option<Value>,
```

we don't need to declare this right?